DiscordAlertsTrader/brokerages/ibkr_api.py

```diff
@@ -0,0 +1,525 @@
+        elif action == "STO":
+            kwargs['action'] = "SELL"
+        Symbol = self.fix_symbol(Symbol, "in")
+
+        if (("_" in Symbol) or (" " in Symbol)):
+            kwargs['asset'] = 'OPT'
+
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol
+            
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty  
+        kwargs['orderType'] = 'LMT' 
+        kwargs['lmtPrice'] = price
+        kwargs['conId'] = self.get_con_id(Symbol)
+
+        return kwargs if kwargs['conId'] is not None else None
+
+    def make_Lim_SL_order(self, Symbol:str, Qty:int,  PT:float, SL:None, action="STC", **kwarg):
+        """Sell with a limit order and a stop loss order"""        
+        kwargs = {}
+        if action == "STC":
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
+
+        Symbol = self.fix_symbol(Symbol, "in")
+        if (("_" in Symbol) or (" " in Symbol)):
+            kwargs['asset'] = 'OPT'
+
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol
+
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty 
+        kwargs['conId'] = self.get_con_id(Symbol)
+        if SL is not None:
+            print("WARNING: webull api does not support OCO, setting a SL only, do not provide SL to send PT")
+            kwargs['orderType'] = 'STP' 
+            kwargs['lmtPrice'] = SL
+        else:
+            print("WARNING: webull api does not support OCO, sending PT without SL")
+            kwargs['orderType'] = 'LMT' 
+            kwargs['lmtPrice'] = PT
+        return kwargs if kwargs['conId'] is not None else None
+
+    def make_STC_lim(self, Symbol:str, Qty:int, price:float, strike=None, action="STC", **kwarg):
+        """Sell with a limit order and a stop loss order"""        
+        kwargs = {}
+        if action == "STC":
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
+
+        Symbol = self.fix_symbol(Symbol, "in")
+        if "_" in Symbol:
+            kwargs['asset'] = 'OPT'
+            
+        else:
+            kwargs['asset'] ='STK'
+            kwargs['outsideRegularTradingHour'] = True
+            kwargs['stock'] = Symbol
+
+        kwargs['enforce'] ='GTC'
+        kwargs['quant'] = Qty  
+        kwargs['orderType'] = 'LMT' 
+        kwargs['lmtPrice'] = price
+        kwargs['conId'] = self.get_con_id(Symbol)
+
+        return kwargs if kwargs['conId'] is not None else None
+
+    def make_STC_SL(self, Symbol:str, Qty:int, SL:float, action="STC", **kwarg):
+        """Sell with a stop loss order"""
+        kwargs = {}
+        if action == "STC":
+            kwargs['action'] = "SELL"
+        elif action == "BTC":
+            kwargs['action'] = "BUY"
```

Comment:
IBRK does not have smth else like "BUYCLOSE"?

Reply:
No actions can only take 'BUY' or 'SELL' as parameters. However, there is Limit on Close and Market on Close order types. Is that what you are looking for here?

Reply:
Reviewing code of other brokers, I don't think this is what you are looking for but please review and let me know 
https://interactivebrokers.github.io/tws-api/basic_orders.html#market_on_close
https://interactivebrokers.github.io/tws-api/basic_orders.html#limitonclose

Reply:
ok just making sure